app/src/main/java/com/google/samples/apps/sunflower/GardenFragment.kt

```diff
@@ -52,8 +52,7 @@ class GardenFragment : Fragment() {
         })
 
         viewModel.plantAndGardenPlantings.observe(viewLifecycleOwner, Observer { result ->
-            if (result != null && result.isNotEmpty())
-                adapter.submitList(result)
+            adapter.submitList(result.orEmpty())
```

Comment:
origin consumption: The logical operation 
after-change consumption:  Null or an empty list will continue to be called deep, consuming the runtime.

Take a look the internal what will happen:

```java
    public void submitList(@Nullable final List<T> newList) {
                     .......
              if (newList == null) {
                      mReadOnlyList = Collections.unmodifiableList(newList);
                      // notify last, after list is updated
                      mUpdateCallback.onInserted(0, newList.size());
                }
            .......
     }
```
I still prefer the origin which demonstrates to avoid submiting list clearly under conditations, although the `result.orEmpty()` is  a really comfortable method. 👍

Reply:
I expected the empty list to be an edge case which doesn't happen too often. I most of the cases a list of items would be passed to adapter. Further action can be limited by using DiffUtil.